GitCommands/Settings/ConfigFileSettings.cs

```diff
@@ -7,45 +7,56 @@ namespace GitCommands.Settings
 {
     public sealed class ConfigFileSettings : SettingsContainer<ConfigFileSettings, ConfigFileSettingsCache>, IConfigFileSettings, IConfigValueStore
     {
-        public ConfigFileSettings(ConfigFileSettings? lowerPriority, ConfigFileSettingsCache settingsCache,
-            SettingLevel settingLevel)
+        private ConfigFileSettings(ConfigFileSettings? lowerPriority, ConfigFileSettingsCache settingsCache, SettingLevel settingLevel)
             : base(lowerPriority, settingsCache)
         {
             SettingLevel = settingLevel;
         }
 
-        public static ConfigFileSettings CreateEffective(GitModule module)
+        public static ConfigFileSettings CreateEffective(GitModule module, bool allowCache = true)
         {
-            return CreateLocal(module, CreateGlobal(CreateSystemWide()), SettingLevel.Effective);
+            return CreateLocal(module,
+                lowerPriority: CreateGlobal(lowerPriority: CreateSystemWide(allowCache), allowCache),
+                SettingLevel.Effective,
+                allowCache);
+        }
+
+        /// <summary>
+        /// Factory added just for accessing the ctor by GitModule.LocalConfigFile in its very special way.
+        /// </summary>
+        public static ConfigFileSettings CreateLocal(ConfigFileSettingsCache effectiveSettingsCache)
+        {
+            return new ConfigFileSettings(lowerPriority: null, effectiveSettingsCache, SettingLevel.Local);
         }
```

Comment:
I'm having difficulties reconciling the xml-doc with the implementation - e.g., there's no obvious correlation between the implementation and `GitModule.LocalConfigFile`. Also, the summary should generally be about what the method does.
Also, is there a reason why this method is public and not internal?

Reply:
I have declared the `ConfigFileSettings` ctor `private` in order to avoid misuse / wrong use in the future (and to find all usages).
To be honest, I do not fully understand the implementation of `GitModule.LocalConfigFile`. So I added this factory method with the same implementation as before.
`GitModule.LocalConfigFile` is unrelated to the original issue, and I hesitate to change this heavily used property.

Reply:
Can we make this `internal`?

Reply:
Done